libsql-sqlite3/src/vector.c

```diff
@@ -66,6 +66,27 @@ Vector *vectorAlloc(VectorType type, VectorDims dims){
   return pVector;
 }
 
+/*
+** Initialize a static Vector object.
+**
+** Note that that the vector object points to the blob so if
+** you free the blob, the vector becomes invalid.
+**/
+void vectorInitStatic(Vector *pVector, VectorType type, const unsigned char *pBlob, size_t nBlobSize){
+  switch (type) {
+    case VECTOR_TYPE_FLOAT32:
+      vectorF32InitFromBlob(pVector, pBlob, nBlobSize);
+      break;
+    case VECTOR_TYPE_FLOAT64:
+      vectorF64InitFromBlob(pVector, pBlob, nBlobSize);
+      break;
+    default:
+      assert(0);
+  }
+  pVector->type = type;
+  pVector->flags = VECTOR_FLAGS_STATIC;
```

Comment:
Wouldn't `pVector->flags |= VECTOR_FLAGS_STATIC` be better?

Reply:
Or is the assumption here that pVector points to blank memory?

Reply:
Yes, `init` will initialize `pVector` from scratch. It always should called as a constructor (it can actually reset the `pVector` state but this must be done carefully in order to not leak resources)

Anyway - so far I suggest to keep explicit assignment (still, we don't have any other flags for now)

Reply:
If it's fresh memory then assignment is better